tedana/metrics/dependence.py

```diff
@@ -0,0 +1,354 @@
+            signal_FT2_Z, noise_FT2_Z, equal_var=False)
+        signal_minus_noise_z[i_comp] = t_to_z(t_value, dof)
+
+    signal_minus_noise_z[np.isnan(signal_minus_noise_t)] = 0
+    signal_minus_noise_p[np.isnan(signal_minus_noise_p)] = 0
+    return signal_minus_noise_z, signal_minus_noise_p
+
+
+def compute_signal_minus_noise_t(Z_maps, Z_cl_maps, F_T2_maps, z_thresh=1.95):
+    RepLGR.info('A t-test was performed between the distributions of T2*-model '
+                'F-statistics associated with clusters (i.e., signal) and '
+                'non-cluster voxels (i.e., noise) to generate a t-statistic '
+                '(metric signal-noise_t) and p-value (metric signal-noise_p) '
+                'measuring relative association of the component to signal '
+                'over noise.')
+    n_components = Z_maps.shape[1]
+    signal_minus_noise_t = np.zeros(n_components)
+    signal_minus_noise_p = np.zeros(n_components)
+    for i_comp in range(n_components):
+        # index voxels significantly loading on component but not from clusters
+        comp_noise_sel = ((np.abs(Z_maps[:, i_comp]) > z_thresh) &
+                          (Z_clmaps[:, i_comp] == 0))
+        comptable.loc[i_comp, 'countnoise'] = np.array(
+            comp_noise_sel, dtype=np.int).sum()
+        # NOTE: Why only compare distributions of *unique* F-statistics?
+        noise_FT2_Z = np.log10(np.unique(F_T2_maps[comp_noise_sel, i_comp]))
+        signal_FT2_Z = np.log10(np.unique(
+            F_T2_maps[Z_clmaps[:, i_comp] == 1, i_comp]))
+        (signal_minus_noise_t[i_comp],
+         signal_minus_noise_p[i_comp]) = stats.ttest_ind(
+             signal_FT2_Z, noise_FT2_Z, equal_var=False)
+
+    signal_minus_noise_t[np.isnan(signal_minus_noise_t)] = 0
+    signal_minus_noise_p[np.isnan(signal_minus_noise_p)] = 0
+    return signal_minus_noise_t, signal_minus_noise_p
+
+
+def compute_dice(Br_clmaps, F_clmaps):
+    n_components = Br_clmaps.shape[1]
+    dice_values = np.zeros(n_components)
+    for i_comp in range(n_components):
+        dice_values[i_comp] = utils.dice(Br_clmaps[:, i_comp], F_clmaps[:, i_comp])
+
+    dice_values[np.isnan(dice_values)] = 0
+    return dice_values
+
+
+def generate_decision_table_score(kappa, dice_FT2, signal_minus_noise_t,
+                                  countnoise, countsigFT2):
+    d_table_rank = np.vstack([
+        len(kappa) - stats.rankdata(kappa),
+        len(kappa) - stats.rankdata(dice_FT2),
+        len(kappa) - stats.rankdata(signal_minus_noise_t),
+        stats.rankdata(countnoise),
+        len(kappa) - stats.rankdata(countsigFT2)]).T
+    d_table_score = d_table_rank.mean(axis=1)
+    return d_table_score
+
+
+def calculate_z_maps(weights, z_max=8):
+    n_components = weights.shape[1]
+    Z_maps = np.zeros(weights.shape)
+    for i_comp in range(n_components):
+        # compute weights as Z-values
+        weights_z = (weights[:, i_comp] - weights[:, i_comp].mean()) / weights[:, i_comp].std()
```

Comment:
Why not `scipy.stats.zscore` instead? Also, should we be using `ddof=1` for the standard deviation calculation?

Reply:
I don't think this step should even be done. `computefeats2` should return z-values (Fisher's R-to-Z performed on parameter estimates like they were Pearson r-values), but not z-statistics. To get from the Fisher's R-to-Z values to test statistics, we would then divide by sqrt(n-3), rather than z-scoring the resulting values. Although, again, that requires r-values going in and not parameter estimates from a _multiple_ regression.